x-pack/legacy/plugins/apm/public/services/rest/callApi.ts

```diff
@@ -22,8 +23,10 @@ function fetchOptionsWithDebug(fetchOptions: FetchOptions) {
 
   const isGet = !fetchOptions.method || fetchOptions.method === 'GET';
 
+  const { body, ...rest } = fetchOptions;
+
   // Need an empty body to pass route validation
-  const body = isGet
+  const bodyObject = isGet
```

Comment:
Why should bodies in GET requests not be stringified?

btw. What do you think about a small rename:

```suggestion
  const bodyObject = isGetRequest
```

Reply:
AFAIR, `body` is not allowed in GET requests per NP requirements. All of this is to work around schema limitations, so hopefully we can get rid of it when https://github.com/elastic/kibana/issues/50179 is addressed.

Reply:
> AFAIR, body is not allowed in GET requests per NP requirements

In that case we'll never send `body` with GET requests, right? So having this check won't matter afaict.

Reply:
Eg. we could always just `JSON.stringify(body)` afaict - or will that cause validation issues?

Reply:
I might not be understanding what you mean. This code adds `{ body: { ... } } ` to all non-GET requests (either empty or whatever it was given). We need at least an empty `body` value to pass the current NP request validation. Do you see a way we can do this without checking the request method?

Reply:
I was thinking something like this. So not depending in the http verb but only whether there is a body or not:
```ts
  return {
    ...rest,
    ...(fetchOptions.body ? JSON.stringify(body) : {}),
    query: {
      ...fetchOptions.query,
      ...(debugEnabled ? { _debug: true } : {})
    }
```

But if NP requires a `body` for all requests then that's not possible (although that sounds like an odd requirement)